### PR TITLE
Yixiao taskeditthunks unit

### PR DESCRIPTION
**Pull request by JYXiao-2021**
# Description
Create the unit test for the thunks component to confirm whether it working as intended.

## Main changes explained:
- -Create file thunks.test.js to do the unit test.

## How to test:
1. check into current branch
2. do npm install to run this PR locally
3. npm test src/components/TaskEditSuggestions/`__tests__`/thunks.test.js

## Screenshots or videos of changes:
image

### CODE REVIEW

File: src/components/TaskEditSuggestions/__tests__/thunks.test.js
Comment (xmyoot) [inline src/components/TaskEditSuggestions/__tests__/thunks.test.js:1]: Hi JYXiao-2021, great work on testing the dispatch of actions related to TaskEditSuggestsions, the only suggestion I could make is maybe separating the fetchTaskEditSuggestionCount into its own file

**Comment by xmyoot** [comment]
Hello, I ran and checked that the tests are running properly, I left a small comment too, great work on the test suite, I learned alot from your code, reference my photo: 
image

**Review by 666saofeng — APPROVED** [review]
All tests passed! 
image

**Review by X1aoZhang99 — APPROVED** [review]
All tests passed. Nice work! 

Screenshot 2024-02-04 at 18 07 28

**Review by peterson337 — APPROVED** [review]
All Tests passed. Good job!

a

**Comment by iTvX** [comment]
Good job, all tests pass!

CleanShot 2024-02-06 at 00 18 37@2x

**Review by ShengjieMao — APPROVED** [review]
Hi, all tests passed. 
I can see all components been rendered, and potential errors/edge cases are tested and their corresponding response are considered. 
great work
PR_1920

**Review by vikrambadhan — APPROVED** [review]
Hi, I tested your PR and all the test cases passed!

Week 24 - PR1

**Review by prnvjn — APPROVED** [review]
All 5 tests were cleared! Test file was easy to understand
Screenshot 2024-02-08 at 6 00 45 PM

**Review by n0vay — APPROVED** [review]
Looks good to me

**Comment by mianmiantea2019** [comment]
All tests passed. Great work!

image

**Review by Pandani07 — APPROVED** [review]
Tested this PR and all the test cases passed. Performed additional bug checking, everything works great, did not run into any errors

Screenshot 2024-02-09 at 5 35 25 PM

**Review by Renanluizssx — APPROVED** [review]
Hi @JYXiao-2021, i tested your unit test and application and it's all correct
Screenshot 2024-02-10 011731

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/91035340/28131a4a-ccc2-4e56-b9e8-fef2f324ad11

**Review by yaow62023 — APPROVED** [review]
Hi, I have reviewed your unit testing, all the test are passed.
 
Screenshot 2024-02-10 at 12 36 24 AM

**Comment by DragonSlayer2401** [comment]
All tests passed as intended. Please watch the overview video below for more details:


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/102876103/efde282f-9862-4108-a010-1fd0b074eb3d

**Review by ClemarNunes — APPROVED** [review]
I tested your PR and all tests passed.

Captura de tela 2024-02-11 060429

**Review by navneeeth — APPROVED** [review]
Hello, @JYXiao-2021! Approving this PR as it comprehensively covers all aspects of the `src/components/TaskEditSuggestions/thunks.js` implementation.

### Test Cases Result
test cases passing
The test cases are all passing for the relevant implementation.

### Review of `src/components/TaskEditSuggestions/__tests__/thunks.test.js`
In the test implementation, great approach in defining the initial state along with setting up a `mockStore` const to simulate a backend server. The tests cover each thunk const in the relevant file:
```javascript
fetchTaskEditSuggestions
rejectTaskEditSuggestion
fetchTaskEditSuggestionCount
```  
I liked how you covered the `dispatches error action on failed HTTP request` case in your unit tests in order to cover the dispatched thunk in that case (which is covered by the catch in the original implementation).

Great implementation! LGTM!

**Comment by one-community** [comment]
Thank you all, merging!

**Pull request merged**